api/end_device.proto

```diff
@@ -175,18 +176,30 @@ message EndDeviceVersion {
 }
 
 message MACSettings {
-  bool use_adr = 1 [(gogoproto.customname) = "UseADR"];
+  google.protobuf.BoolValue use_adr = 1 [(gogoproto.customname) = "UseADR"];
```

Comment:
Although this doesn't break existing JSON API, this is not binary compatible with our existing protos, so existing databases and existing clients/servers may become incompatible.

Reply:
Also, I don't know how the CLI's flag handling will work for this. We may have to write some logic for these wrappers (I can do that if needed). Also below